File: api/src/Domain/ProductCatalog/Pads/DataTransferObjects/PadCreateOrUpdate.cs
using System.Collections.Generic;

namespace DetailingArsenal.Domain.ProductCatalog {
    public class PadCreateOrUpdate : IDataTransferObject {
        public string Name { get; set; }
        public PadCategory Category { get; set; }
        public PadMaterial Material { get; set; }
        public PadTexture Texture { get; set; }
        public List<PolisherType> PolisherTypes { get; set; }
        public DataUrlImage? Image { get; set; }

        public PadCreateOrUpdate() {
            // Used by API layer

            Name = null!;
            PolisherTypes = new List<PolisherType>();
        }

        public PadCreateOrUpdate(string name, PadCategory category, PadMaterial material, PadTexture texture, List<PolisherType> polisherTypes, DataUrlImage? image) {
            Name = name;
            Category = category;
            Material = material;
            Texture = texture;
            PolisherTypes = polisherTypes;
            Image = image;
        }
    }
}

File: app/src/api/product-catalog/data-transfer-objects/pad-create-or-update.ts
import { PadCategory } from '@/api';
import { PadMaterial } from './pad-material';
import { PadTexture } from './pad-texture';
import { PolisherType } from './polisher-type';
import { Image } from './image';

export interface PadCreateOrUpdate {
    name: string,
    category: PadCategory,
    material: PadMaterial,
    texture: PadTexture,
    polisherTypes: PolisherType[],
    image: Image
}